### PR TITLE
Added Edit History Plugin

### PR DESCRIPTION
**Pull request by antoniotejada**
Edit History plugin https://github.com/antoniotejada/obsidian-edit-history

Please switch to **Preview** and select one of the following links:

* [Community Plugin](?template=plugin.md)

### CODE REVIEW

**Comment by joethei** [comment]
Please follow the pull request template.

**Comment by antoniotejada** [comment]
Thanks @joethei , following the instructions ("switch to preview and select...") didn't bring the template. I have now manually gone to the .github directory and copied the raw md below.

# I am submitting a new Community Plugin

## Repo URL

Link to my plugin:

Edit History plugin https://github.com/antoniotejada/obsidian-edit-history

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

**Comment by joethei** [comment]
Why do you have a `releases` folder in your repo?
The releases section on GH offers the same functionality, you should remove that folder from the repo.

["description": "Edit history plugin for Obsidian, automatically saves the history of edits of a file when Obsidian saves the file, and allows viewing the differences between edits, copying text from a previous edit, or fully rolling back to a previous edit.",](https://github.com/antoniotejada/obsidian-edit-history/blob/78c8b4aa613bae7719cd3737975ba91560ad59a1/manifest.json#L6)
We have recently published a guide for plugin descriptions to follow here: https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple.
Avoid including sentences like "This is a plugin for Obsidian" in your description.

[console.log("parseSettings");](https://github.com/antoniotejada/obsidian-edit-history/blob/78c8b4aa613bae7719cd3737975ba91560ad59a1/main.ts#L134)
Please avoid unnecessary logging. If you need it for debugging purposes, add a check to only log during development.

[id: "save-edit-history",](https://github.com/antoniotejada/obsidian-edit-history/blob/78c8b4aa613bae7719cd3737975ba91560ad59a1/main.ts#L526)
Obsidian will automatically prefix the command id with the plugin id,
you don't need to include it yourself, can just shorten the id here to `save`

[name: "Save current edit in the Edit History",](https://github.com/antoniotejada/obsidian-edit-history/blob/78c8b4aa613bae7719cd3737975ba91560ad59a1/main.ts#L527)
In the command pallete this will show up as `Edit History: Save current edit in the Edit History`
You can shorten this to be `Save current edit`

[// XXX Move these to styles.css](https://github.com/antoniotejada/obsidian-edit-history/blob/78c8b4aa613bae7719cd3737975ba91560ad59a1/main.ts#L574)
Please do this with all `.style` calls.

[containerEl.createEl("h3", {text: "Edit History Settings"});](https://github.com/antoniotejada/obsidian-edit-history/blob/78c8b4aa613bae7719cd3737975ba91560ad59a1/main.ts#L770)
[Only use headings under settings if you have more than one section. ](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)
[Avoid "settings" in settings headings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%22settings%22+in+settings+headings)

[let author = containerEl.createEl("small", { text: "Created by "});](https://github.com/antoniotejada/obsidian-edit-history/blob/78c8b4aa613bae7719cd3737975ba91560ad59a1/main.ts#L772-L774)
Would recommend moving this to the end.

[.setDesc("Show Edit History File information on the status bar. Click the status bar to show the Edit History for the current file.")](https://github.com/antoniotejada/obsidian-edit-history/blob/78c8b4aa613bae7719cd3737975ba91560ad59a1/main.ts#L778)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

**Comment by antoniotejada** [comment]
I pushed a new version 0.1.1 https://github.com/antoniotejada/obsidian-edit-history/releases/tag/0.1.1

> Why do you have a `releases` folder in your repo? The releases section on GH offers the same functionality, you should remove that folder from the repo.

I prefer to do it this way (I have the release history in my local git, not just on github, and if I move away from github I don't lose it). In addition, that shouldn't affect you since you won't be pulling the sources.

> ["description": "Edit history plugin for Obsidian, automatically saves the history of edits of a file when Obsidian saves the file, and allows viewing the differences between edits, copying text from a previous edit, or fully rolling back to a previous edit.",](https://github.com/antoniotejada/obsidian-edit-history/blob/78c8b4aa613bae7719cd3737975ba91560ad59a1/manifest.json#L6) We have recently published a guide for plugin descriptions to follow here: https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple. Avoid including sentences like "This is a plugin for Obsidian" in your description.

Done

> [console.log("parseSettings");](https://github.com/antoniotejada/obsidian-edit-history/blob/78c8b4aa613bae7719cd3737975ba91560ad59a1/main.ts#L134) Please avoid unnecessary logging. If you need it for debugging purposes, add a check to only log during development.

I added a suite of logging overrides and a dropdown to select the debug level in the settings. 

> [id: "save-edit-history",](https://github.com/antoniotejada/obsidian-edit-history/blob/78c8b4aa613bae7719cd3737975ba91560ad59a1/main.ts#L526) Obsidian will automatically prefix the command id with the plugin id, you don't need to include it yourself, can just shorten the id here to `save`
> 
> [name: "Save current edit in the Edit History",](https://github.com/antoniotejada/obsidian-edit-history/blob/78c8b4aa613bae7719cd3737975ba91560ad59a1/main.ts#L527) In the command pallete this will show up as `Edit History: Save current edit in the Edit History` You can shorten this to be `Save current edit`

Removing those makes it harder to understand, I'm leaving those in, but I changed the capitalization to match your request below.

> [// XXX Move these to styles.css](https://github.com/antoniotejada/obsidian-edit-history/blob/78c8b4aa613bae7719cd3737975ba91560ad59a1/main.ts#L574) Please do this with all `.style` calls.

Done

> [containerEl.createEl("h3", {text: "Edit History Settings"});](https://github.com/antoniotejada/obsidian-edit-history/blob/78c8b4aa613bae7719cd3737975ba91560ad59a1/main.ts#L770) [Only use headings under settings if you have more than one section. ](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.) [Avoid "settings" in settings headings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%22settings%22+in+settings+headings)

I removed the main title and reorganized in sections.

> [let author = containerEl.createEl("small", { text: "Created by "});](https://github.com/antoniotejada/obsidian-edit-history/blob/78c8b4aa613bae7719cd3737975ba91560ad59a1/main.ts#L772-L774) Would recommend moving this to the end.

I'm leaving that there for the time being.

> [.setDesc("Show Edit History File information on the status bar. Click the status bar to show the Edit History for the current file.")](https://github.com/antoniotejada/obsidian-edit-history/blob/78c8b4aa613bae7719cd3737975ba91560ad59a1/main.ts#L778) [Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

I see Edit History and Edit History File as proper nouns, and that made text easier to understand, but since I wasn't being consistent across, I just put it the way you want it.
